logging/wagclientlogger/nooplogger.go

```diff
@@ -0,0 +1,7 @@
+package wagclientlogger
+
+type NoOpLogger struct{}
```

Comment:
We should have had one of these a long time ago.

Reply:
yeah I think this would be worth cherry-picking into its own PR, probably?

Reply:
Agreed